server/service/osquery.go

```diff
@@ -422,6 +422,30 @@ func (svc *Service) GetClientConfig(ctx context.Context) (map[string]interface{}
 		config["packs"] = json.RawMessage(packJSON)
 	}
 
+	// Get all scheduled queries for the current team
+	scheduledQueries, err := svc.ds.ListQueries(ctx, fleet.ListQueryOptions{
+		TeamID:      host.TeamID,
+		IsScheduled: ptr.Bool(true),
+	})
+	scheduledConfig := make(fleet.Queries, len(scheduledQueries))
+	for _, query := range scheduledQueries {
+		scheduledConfig[query.Name] = fleet.QueryContent{
+			Query:    query.Query,
+			Interval: query.ScheduleInterval,
+			Platform: &query.Platform,
+			Version:  &query.MinOsqueryVersion,
+			Removed:  query.GetRemoved(),
+			Snapshot: query.GetSnapshot(),
+		}
+	}
+	if len(scheduledConfig) > 0 {
+		payload, err := json.Marshal(scheduledConfig)
+		if err != nil {
+			return nil, newOsqueryError("internal error: marshal schedule JSON: " + err.Error())
+		}
+		config["schedule"] = json.RawMessage(payload)
```

Comment:
This is a behavior change, in the current version we send all scheduled queries under `config["packs"]`, there's one pack for global, one pack per team, and then user packs and all are set in `config["packs"]`.

I wonder if we should stick to the old behavior or use this new behavior. There's some risk but if osquery behaves as expected then it should just work.

Reply:
So scheduled queries should be added either to `config["packs"]["Global"]` (if `TeamID = nil`) or to `config["packs"]["team-$ID"]` depending on whether the host belongs to a team?

Reply:
Here's a sample:
```json
{
  [...]
  "packs": {
    "CMMC Level 1 (MacOS)": {
      "queries": {}
    },
    "Global": {
      "queries": {
        "Get USB device type": {
          "query": "SELECT vendor, model FROM usb_devices WHERE removable = 1;",
          "interval": 604800,
          "platform": "",
          "version": "",
          "snapshot": true,
          "removed": false
        },
        "Get recently added or removed USB drives": {
          "query": "SELECT action, DATETIME(time, 'unixepoch') AS datetime, vendor, mounts.path \nFROM disk_events \nLEFT JOIN mounts \n    ON mounts.device = disk_events.device\n;",
          "interval": 86400,
          "platform": "",
          "version": "",
          "snapshot": true,
          "removed": false
        }
      }
    },
    "Malware": {
      "queries": {}
    },
    "Security Tooling Checks": {
      "queries": {}
    },
    "Team: 📊 CIS Benchmarks": {
      "queries": {}
    },
    "g-testpack": {
      "queries": {}
    }
  }
}
```

Reply:
(Reminder that global schedules are inherited, so all hosts should get them.)

Reply:
Done in https://github.com/fleetdm/fleet/pull/12723/commits/5428dc6ceab305ed712e67a707030e8a6cc869a1